cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -2236,10 +2369,166 @@ def do_setup(self, context):
+            'vmware_profile_name': profile,
+        }
+
+        return provider_info
+
+    @utils.trace
+    def _get_datastores_for_profiles(self):
+        datastores = {}
+        for profile in self._storage_profiles:
+            for h in pbm.filter_hubs_by_profile(self.session,
+                                                None,
+                                                profile['id']):
+                if h.hubType != "Datastore":
+                    # We are not interested in Datastore Clusters for now
+                    continue
+                if h.hubId not in datastores:
+                    # Reconstruct a managed object reference to that
+                    # datastore
+                    ds = vim_util.get_moref(h.hubId, "Datastore")
+                    summary = self.volumeops.get_summary(ds)
+                    datastores[summary.name] = {'summary': summary,
+                                                'storage_profile': profile}
+
+        return datastores
+
+    def _new_host_for_volume(self, volume):
+        pool_info = volume_utils.extract_host(
+            volume.host, level='pool', default_pool_name=True)
+        model = self._volume_provider_metadata(volume)
+        if pool_info != model['vmware_ds_name']:
+            host = volume_utils.extract_host(
+                volume.host, level='host')
+            back = volume_utils.extract_host(volume.host)
+            backend = back.split('@')[1]
+
+            new_host = '{}@{}#{}'.format(
+                host, backend,
+                model['vmware_ds_name']
+            )
+            return new_host
+
+    @utils.trace
+    def update_provider_info(self, volumes, snapshots):
+        """Ensure we have a provider_id set on volumes.
+
+        If there is a provider_id already set then skip, if it is missing then
+        we will update it based on the volume object. We can always compute
+        the id if we have the full volume object, but not all driver API's
+        give us that info.
+
+        We have to save each volume entry if they update their host, otherwise
+        cinder volume manager doesn't see that volume as part of the host
+        allocated_capacity calculation, which happens right after this call
+        completes.
+
+        We don't care about snapshots, they just use the volume's provider_id.
+        """
+        LOG.info("HOST {} : volumes {}".format(self.host, len(volumes)))
+        if self.configuration.vmware_datastores_as_pools:
+            LOG.info("vmware_datastores_as_pools is enabled. "
+                     "Checking host entries for volumes and snapshots.")
+            datastores = self._get_datastores_for_profiles()
+            ds_keys = datastores.keys()
+            vol_updates = []
+            LOG.info("Process {} volumes".format(len(volumes)))
+            for vol in volumes:
+                # make sure we have the correc host info
+                if vol['status'] in ['in-use', 'available']:
```

Comment:
Missing a `t` in the comment.

I'd turn this condition around again and use `continue` to get less indentation.